File: sdk/nodejs-sdk/src/protocols/v1_0/OutOfBand.js
'use strict'
const utils = require('../../utils')
const Protocol = require('../Protocol')
/*
 * NON_VISIBLE
 *
 * This is an implementation of OutOfBandImplV1_0 but is not viable to user of Verity SDK. Created using the
 * static PresentProof class
 * @extends Protocol
 */
class OutOfBand extends Protocol {
  /**
     * Creates an OutOfBand Protocol
     * @param forRelationship the relationship identifier (DID) for the pairwise relationship that will be used
     * @param inviteUrl the Out-of-Band invitation url
     * @return OutOfBand object
     *
     * @property {String} msgFamily - 'out-of-band'
     * @property {String} msgFamilyVersion - '1.0'
     * @property {String} msgQualifier - 'Community Qualifier'
     * @property {String} this.msgNames.REUSE - 'reuse'
     */
  constructor (forRelationship = null, inviteUrl = null) {
    const msgFamily = 'out-of-band'
    const msgFamilyVersion = '1.0'
    const msgQualifier = utils.constants.COMMUNITY_MSG_QUALIFIER
    super(msgFamily, msgFamilyVersion, msgQualifier)

    this.msgNames.REUSE = 'reuse'
    this.msgNames.RELATIONSHIP_REUSED = 'relationship-reused'

    this.forRelationship = forRelationship
    this.inviteUrl = inviteUrl
  }

  /**
  * Creates the control message without packaging and sending it.
  *
  * @param context an instance of the Context object initialized to a verity-application agent
  * @return the constructed message (JSON object)
  *
  * @see #handshakeReuse
  */
  async reuseMsg (context) {
    var msg = this._getBaseMessage(this.msgNames.REUSE)
    msg['~for_relationship'] = this.forRelationship
    msg.inviteUrl = this.inviteUrl
    msg = this._addThread(msg)
    return msg
  }

  /**
  * Creates and packages message without sending it.
  *
  * @param context an instance of the Context object initialized to a verity-application agent
  * @return the byte array ready for transport
  *
  * @see #handshakeReuse
  */
  async reuseMsgPacked (context) {
    return this.getMessageBytes(context, await this.reuseMsg(context))
  }

  /**
  * Direct the verity-application agent to reuse the relationship given.
  *
  * @param context an instance of the Context object initialized to a verity-application agent
  */
  async reuse (context) {
    await this.sendMessage(context, await this.reuseMsgPacked(context))
  }
}
module.exports = OutOfBand

File: sdk/python-sdk/verity_sdk/protocols/v1_0/OutOfBand.py
from decimal import Context

from verity_sdk.protocols.Protocol import Protocol
from verity_sdk.utils import COMMUNITY_MSG_QUALIFIER


class OutOfBand(Protocol):
    """
    The OutOfBand protocol allow an interaction intent, connection information and reuse of a relationship.
    """

    MSG_FAMILY = 'out-of-band'
    """the family name for the message family"""
    MSG_FAMILY_VERSION = '1.0'
    """the version for the message family"""

    REUSE = 'reuse'
    """Name for 'reuse' control message"""

    def __init__(self,
                 for_relationship: str = None,
                 invite_url: str = None):
        """
        Args:
            for_relationship (str): the relationship identifier (DID) for the pairwise relationship that will be reused
            invite_url (str): the Out-of-Band invitation url
        """
        super().__init__(
            self.MSG_FAMILY,
            self.MSG_FAMILY_VERSION,
            msg_qualifier=COMMUNITY_MSG_QUALIFIER
        )
        self.for_relationship = for_relationship
        self.invite_url = invite_url

    async def reuse(self, context: Context):
        """
        Direct the verity-application agent to reuse the relationship given.

        Args:
            context (Context): an instance of the Context object initialized to a verity-application agent
        """
        await self.send_message(context, await self.reuse_msg_packed(context))

    def reuse_msg(self, context: Context):
        """
        Creates the control message without packaging and sending it.

        Args:
            context (Context): an instance of the Context object initialized to a verity-application agent

        Return:
            the constructed message (dict object)
        """
        msg = self._get_base_message(self.REUSE)
        self._add_thread(msg)
        self._add_relationship(msg, self.for_relationship)

        msg['inviteUrl'] = self.invite_url
        return msg

    async def reuse_msg_packed(self, context: Context):
        """
        Creates and packages message without sending it.

        Args:
            context (Context): an instance of the Context object initialized to a verity-application agent

        Return:
            the bytes ready for transport
        """
        return await self.get_message_bytes(context, self.reuse_msg(context))
